plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -226,18 +284,98 @@ public void testMigrateUnsupportedColumnType()
         String hiveTableName = "hive.tpch." + tableName;
         String icebergTableName = "iceberg.tpch." + tableName;
 
-        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
+        assertUpdate("CREATE TABLE " + hiveTableName + " AS SELECT tinyint '1' x", 1);
+        assertQuery("SELECT * FROM " + hiveTableName, "SELECT 1");
 
-        assertQueryFails(
-                "CALL iceberg.system.migrate('tpch', '" + tableName + "')",
-                "\\QTimestamp precision (3) not supported for Iceberg. Use \"timestamp(6)\" instead.");
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '%s')".formatted(tableName), "Type not supported for Iceberg: tinyint");
+
+        assertQuery("SELECT count(*) FROM " + hiveTableName, "SELECT 1");
+        assertQueryFails("SELECT count(*) FROM " + icebergTableName, "Not an Iceberg table: .*");
 
+        assertUpdate("DROP TABLE " + hiveTableName);
+    }
+
+    @DataProvider
+    public static Object[][] timestampPrecisionsAndValuesOrc()
+    {
+        return new Object[][] {
+                {HiveTimestampPrecision.MILLISECONDS, "2021-01-01 06:12:12.123"},
+                {HiveTimestampPrecision.MICROSECONDS, "2021-01-01 06:12:12.123456"}
+        };
+    }
+
+    @Test(dataProvider = "timestampPrecisionsAndValuesOrc")
+    public void testMigrateColumnTypeTimestampOnFormatOrc(HiveTimestampPrecision precision, String timestamp)
+    {
+        String tableName = "test_migrate_timestamp_orc" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+        String icebergTableName = "iceberg.tpch." + tableName;
+        Session session = timestampPrecisionSession(precision);
+
+        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(" + precision.getPrecision() + ")) WITH (format='" + IcebergFileFormat.ORC + "') ");
```

Comment:
No need to use constant here. 
```suggestion
        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(" + precision.getPrecision() + ")) WITH (format='ORC')");
```